tools/run_tests/run_xds_tests.py

```diff
@@ -96,7 +97,11 @@ def parse_port_range(port_arg):
 
 
 argp = argparse.ArgumentParser(description='Run xDS interop tests on GCP')
-argp.add_argument('--project_id', help='GCP project id')
+argp.add_argument('--project_id', default='grpc-testing', help='GCP project id')
+argp.add_argument(
+    '--project_num',
```

Comment:
Not sure if this is ready for review yet, but: please get project number programmatically (https://cloud.google.com/resource-manager/reference/rest/v1/projects/get) rather than requiring specifying both the id and the number on the command-line. I'm also opposed to setting a default value here and would prefer continuing to require explicit specification of the ID.

Reply:
I temporarily add default values and will remove them once all callers provide both project_id and project_num. Both of them will be required args. On the other hand, the getting project number programmatically API need an authentication token. It's not easy to provide it securely. WDYT?

Reply:
I don't think that getting the project number requires a new authentication token. All of our REST API calls to GCE already require authentication, which is set up using the google default creds on the VM. The doc page I linked to says that the caller requires IAM permission `resourcemanager.projects.get` on the project - I think our VMs already have this. I just ssh'ed into one and ran the snippet from the documentation page I linked:

```
resource_manager = googleapiclient.discovery.build('cloudresourcemanager', 'v1') 
project_request = resource_manager.projects().get(projectId='grpc_testing')
project_response = project_request.execute()
>>> pprint(project_response)
{'createTime': ...,
 'projectId': 'grpc-testing',
 'projectNumber': '830293263384'}
```

And it worked as-is. So you would just need to construct the parallel `resource_manager` object analogous to how we construct a `compute` accessor here: https://github.com/grpc/grpc/blob/master/tools/run_tests/run_xds_tests.py#L2009 (we already have a dependency on googleapiclient for use with the GCE rest API).

If the above doesn't work for some reason, then setting both by default until all callers are updated, and then removing the default sounds reasonable. But given the number of calls I think it would be easier to fetch programmatically (there could totally be problems I'm not seeing with authentication, as I don't have a complete grasp of the permissions needed vs how they're setup by Kokoro, but just going by a quick manual trial, it seems we already have the necessary permissions in our CI VMs)

Reply:
I see, the authentication is not a problem using the snippet. I think getting project number from id programmatically is still more complex than providing both project number and id in the command line args.